rllib/env/multi_agent_episode.py

```diff
@@ -755,6 +753,108 @@ def finalize(
 
         return self
 
+    def concat_episode(self, other: "MultiAgentEpisode") -> None:
+        """Adds the given `other` MultiAgentEpisode to the right side of self.
+
+        In order for this to work, both chunks (`self` and `other`) must fit
+        together. This is checked by the IDs (must be identical), the time step counters
+        (`self.t` must be the same as `episode_chunk.t_started`), as well as the
+        observations/infos at the concatenation boundaries (`self.observations[-1]`
+        must match `episode_chunk.observations[0]`). Also, `self.is_done` must
+        not be True, meaning `self.is_terminated` and `self.is_truncated` are both
+        False.
+
+        Args:
+            other: The other `MultiAgentEpisode` to be concatenated to this one.
+
+        Returns: A `MultiAgentEpisode` instance containing the concatenated data
+            from both episodes (`self` and `other`).
+        """
+        # Make sure the IDs match.
+        assert other.id_ == self.id_
+        # NOTE (sven): This is what we agreed on. As the replay buffers must be
+        # able to concatenate.
+        assert not self.is_done
+        # Make sure the timesteps match.
+        assert self.env_t == other.env_t_started
+        # Validate `other`.
+        other.validate()
+
+        # Concatenate the individual SingleAgentEpisodes from both chunks.
+        all_agent_ids = set(self.agent_ids) | set(other.agent_ids)
+        for agent_id in all_agent_ids:
+            sa_episode = self.agent_episodes.get(agent_id)
+
+            # If agent is only in the new episode chunk.
+            if sa_episode is None:
+                # Then store all agent data from the new episode chunk in self.
+                self.agent_episodes[agent_id] = other.agent_episodes[agent_id]
+                # Do not forget the env to agent timestep mapping.
+                self.env_t_to_agent_t[agent_id] = other.env_t_to_agent_t[agent_id]
```

Comment:
Are the `env_t_to_agent_t` already updated with the new `env_t_started` of the `other` or do we have to add the starting point of the `other` to `env_t_to_agent_t` here?

Reply:
They get concatenated as well. Note that - for now - `env_t_to_agent_t` always start at 0, no matter what the env_t_started is. They basically operate on a "simpler" time-axis.
We might want to change this in the future (and use the true global env steps instead). But I'm not sure yet, whether this would not complicate things too much. After all, it's just an int offset that we are talking about here.